packages/eslint-config-react/rules/import.js

```diff
@@ -0,0 +1,25 @@
+module.exports = {
+  overrides: [
+    /**
+     * This override enables default exports for files that require them.
+     * @link https://github.com/reside-eng/guidelines/blob/main/rfcs/shared-patterns/typescript/001-consistent-import-export.md
+     */
+    {
+      files: [
+        // Storybook stories
+        '**/*.stories.{jsx,tsx}',
+
+        // Next.js pages and API routes
+        'src/pages/**/*.{jsx,tsx}',
+        'pages/**/*.{jsx,tsx}',
+
+        // Config files
+        'next.config.{js,mjs}',
+        'jest.config.{js,mjs,ts}',
```

Comment:
Should we also include .spec test files since they shouldn't ever need to have a default export or even an export at all? Was brought up in [this PR adding to db-migration-functions](https://github.com/reside-eng/db-migration-functions/pull/96/files)

Reply:
My understand is that if a file does not make an export, it will not get picked up by the linter.

Reply:
We would only need to add a file glob to this list if the file in question needed to make an explicit default export.

Reply:
Ahh yes, that is right. I got the polarity confused with the other PR noted above

Reply:
No worries! Yeah, the names of the rules are a little hard to grok when they are right next to each other. 

Turning off `import/prefer-default-export` will prevent errors for this:
```ts
// There is only a single module export and it's a named export.
export const foo = 'foo';
```

Turning on `import/no-default-export` will explicitly prohibit default exports (only if they exist in a file):

```ts
// There is a default export.
export const foo = 'foo';
const bar = 'bar';
export default 'bar';
```

Reply:
@prescottprue if you are okay with this configuration, would you be able to resolve this thread? Thank you!

Reply:
@htunnicliff if Jest is needed, then that config should be in base not the react one.

Reply:
Jest is now in the base config.